test/lit_tests/raising/multidim.mlir

```diff
@@ -0,0 +1,21 @@
+// RUN: enzymexlamlir-opt %s --delinearize-indexing --canonicalize --split-input-file | FileCheck %s
+
+// CHECK-LABEL:   func.func private @kern$par0(
+// CHECK-SAME:                                 %[[VAL_0:[^:]*]]: memref<?x5x3xi64, 1>) {
+// CHECK:           %[[VAL_1:.*]] = arith.constant 1 : i64
+// CHECK:           affine.parallel (%[[VAL_2:.*]], %[[VAL_3:.*]], %[[VAL_4:.*]]) = (0, 0, 0) to (7, 5, 3) {
+// CHECK:             affine.store %[[VAL_1]], %[[VAL_0]]{{\[}}%[[VAL_2]] + (%[[VAL_3]] + %[[VAL_4]] floordiv 3) floordiv 5, (%[[VAL_3]] + %[[VAL_4]] floordiv 3) mod 5, %[[VAL_4]] mod 3] : memref<?x5x3xi64, 1>
```

Comment:
the outer one should get simplified to floordiv 15, right?

Reply:
we also ideally want to be able to remove val4 floordiv3 -> 0 [since its always less than 3]